static/js/list_widget.js

```diff
@@ -190,6 +190,24 @@ export function create($container, list, opts) {
         }
     };
 
+    // Used in case of Multiselect DropdownListWidget to retain
+    // previously checked items even after widget redraws.
+    widget.retain_selected_items = function () {
+        const items = opts.multiselect;
+
+        if (items.selected_items) {
+            const data = items.selected_items;
+            for (const value of data) {
+                const list_item = $container.find(`li[data-value = "${value}"]`);
+                if (list_item.length) {
+                    const link_elem = list_item.find("a").expectOne();
+                    list_item.addClass("checked");
+                    link_elem.prepend($("<i>", {class: "fa fa-check"}));
+                }
```

Comment:
This little block is repeated in a few places; I think having a non-exported `mark_option_as_checked` think might be a good idea.

Reply:
Apologies if I misunderstood it. But this block of code is present in `dropdown_list_widget` and `list_widget`. What exactly should be done here to avoid de-duplication between the 2 instances without exporting the function ?

Reply:
I'm not sure; one idea would be to have a `create_list_item` function that is exported.